### PR TITLE
CircleCI integration 

### PR DESCRIPTION
**Pull request by afmicc**
#### Description:

* Add CircleCI configuration

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          command: |
+            mkdir /tmp/test-results
+            TEST_FILES="$(circleci tests glob "spec/**/*_spec.rb" | \
+              circleci tests split --split-by=timings)"
+
+            bundle exec rspec \
+              --out /tmp/test-results/rspec.xml \
+              --format progress \
+              $TEST_FILES
+
```
Comment (mdornel) [inline .circleci/config.yml:68]: I would add a block for code analysis too
```
# run code_analysis!
      - run:
          name: run code analysis
          command: RAILS_ENV=test bundle exec - your code analysis command here! -   
```

**Review by mdornel — APPROVED** [review]
Good job!

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            RAILS_ENV: test
+
+      # Specify service dependencies here if necessary
+      # CircleCI maintains a library of pre-built images
+      # documented at https://circleci.com/docs/2.0/circleci-images/
+      - image: circleci/postgres:10-alpine  # service image
+        environment:
+          POSTGRES_USER: target_api_user
+          POSTGRES_DB: target_api_test
+          POSTGRES_PASSWORD: ""
```
Comment (JuanfraM) [inline .circleci/config.yml]: I think the value `""` is the same as nothing

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+          paths:
+            - ./vendor/bundle
+          key: v1-dependencies-{{ checksum "Gemfile.lock" }}
+
+      # Database setup
+      - run: cp .circleci/database.yml config/database.yml
+      - run: dockerize -wait tcp://localhost:5432 -timeout 1m
+      - run: bundle exec rake db:create
+      - run: bundle exec rake db:schema:load
+      # - run: bundle exec rake db:setup
```
Comment (JuanfraM) [inline .circleci/config.yml]: remove this commented code

**Review by JuanfraM — APPROVED** [review]
Good work!
What @mdornel said is right, the task for code analysis should be here.
You have a new PR with that task, you can merge this PR and then create a new one to add it here, or you can wait and merge this PR after the other one. It's up to you

**Pull request merged**